src/drivers/mysql/drv_mysql.c

```diff
@@ -972,8 +971,34 @@ db_error_t mysql_drv_query(db_conn_t *sb_conn, const char *query, size_t len,
   int err = mysql_real_query(con, query, len);
   DEBUG("mysql_real_query(%p, \"%s\", %zd) = %d", con, query, len, err);
 
-  if (SB_UNLIKELY(err != 0))
-    return check_error(sb_conn, "mysql_drv_query()", query, &rs->counter);
+  if (SB_UNLIKELY(err != 0)) {
+    /* Handle guaranteed capacity error as a separate case.
+     *
+     * When a mysql error occurs, the following things happen
+     * 1. error counter incremented.
+     * 2. cpu_utilization is added.
+     * 3. sb_event_stop not called and therefore
+     * 4. Latency histogram NOT UPDATED (affects latency report)
+     * 5. Overall event counter also not incremented
+     *
+     * 3164 Error = When Audit plugin drops the query
+     * We treat it as a valid query and therefore
+     *
+     * 1. increment error counter + guaranteed_capacity counter
+     * 2. cpu_utilization is added as before
+     * 3. sb_event_stop called and therefore
+     * 4. Latency histogram UPDATED (affects latency report)
+     * 5. Overall event counter also incremented
+     */
+    if (mysql_errno(con) == 3164)
```

Comment:
No constant available for 3164?

Reply:
We should define this as a constant; this might be subject to change. (May want to use custom error range (50-52k https://dev.mysql.com/doc/refman/8.0/en/error-message-components.html#error-code-ranges)